extensions/objects/templates/FractionEditor.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+oppia.directive('fractionEditor', [
+  '$compile', 'FractionObjectFactory', 'OBJECT_EDITOR_URL_PREFIX',
+  function($compile, FractionObjectFactory, OBJECT_EDITOR_URL_PREFIX) {
+    return {
+      link: function(scope, element) {
+        scope.getTemplateUrl = function() {
+          return OBJECT_EDITOR_URL_PREFIX + 'Fraction';
+        };
+        $compile(element.contents())(scope);
+      },
+      restrict: 'E',
+      scope: true,
+      template: '<span ng-include="getTemplateUrl()"></span>',
+      controller: ['$scope', function($scope) {
+        $scope.alwaysEditable = true;
+        $scope.largeInput = false;
```

Comment:
Btw, I think you might want to just go ahead and create a separate HTML file for the fractions template. It is going to be sufficiently different (fractions have validation, accepted characters, etc. and plaintext has large/small input field options) that I think we ought to just have them be different from the start.

Reply:
Yep, that makes sense actually. The new HTML file still mirrors the old one a little as I've left all the validation stuff for the js (just to have some more flexibility in how the errors are treated). Also, do you know what the alwaysEditable flag is for (and if it needs to be in the new HTML file)?

Reply:
Nope, you don't need the alwaysEditable flag. That's a holdover from bygone eras :)

Reply:
You can also drop $scope.largeInput, btw.